### ISSUE TITLE
Drop support for NAPARI_PERFMON with an existing QApplication

### ISSUE THREAD

**Issue by tlambert03** [napari/napari#3166]
## 🧰 Task
I propose we drop support for using `NAPARI_PERFMON` with an existing `QApplication`.

The only real application for this is when you want to use something like `NAPARI_PERFMON=1 ipython --gui qt` and then start a napari viewer.   It then actually deletes the QApplication  so it can create a new one.  This is just super error prone and something we shouldn't try to support.  

Note: removing this doesn't prevent someone from using perfmon within ipython, one just has to create the napari viewer _before_ using the `%gui qt` magic... so we get to create the application ourselves.

Any oppositions to deleting the whole `convert_app_for_tracing` logic?
cc @sofroniewn

**Comment by sofroniewn** [comment]
If we added "perfmon" to our settings now would we ever be able to enable perfmon without a restart or would we always require a restart even with this code? 

It might be nice to have perfmon in settings one day, and if we're ok always requiring a restart then i think it's fine to drop all that logic.

Would be nice to also update the perfmon docs to make it very clear how to support the ipython/ jupyter use case

**Comment by tlambert03** [comment]
> we ever be able to enable perfmon without a restart or would we always require a restart even with this code?

it would always require a restart.  Given the deep nature of the QApplication monkeypatching, it feels acceptable to me to enter in a "profiling session" explicitly.

**Comment by tlambert03** [comment]
> Would be nice to also update the perfmon docs to make it very clear how to support the ipython/ jupyter use case

the only thing we'd need to add is to _not_ use `%gui qt` before starting napari.  It's already not in the docs and gone from most of our examples... but it would also give a warning if someone tried it

**Comment by sofroniewn** [comment]
> it would always require a restart. Given the deep nature of the QApplication monkeypatching, it feels acceptable to me to enter in a "profiling session" explicitly.

ok makes sense, i'm supportive!

> the only thing we'd need to add is to not use %gui qt before starting napari. It's already not in the docs and gone from most of our examples... but it would also give a warning if someone tried it

oh nice, even better!!
